packages/next-auth/src/providers/credentials.ts

```diff
@@ -17,7 +17,7 @@ export interface CredentialsConfig<
   authorize: (
     credentials: Record<keyof C, string> | undefined,
     req: Pick<RequestInternal, "body" | "query" | "headers" | "method">
-  ) => Awaitable<(Omit<User, "id"> | { id?: string }) | null>
```

Comment:
@ThangHuuVu We use SERIAL id's as we have a PostgreSQL database coming from v3, the removal of this line triggers a type error as number is not assignable to string.

Reply:
Do we need to extend the `User` interface as well? @emrysal